Updated Working Group RFC based on feedback (#28)
rust-lang/wg-governance: draft-rfcs/working-group-terminology.md

> While this definition is true, it does not define the difference between what has becomed to be called "Domain Working Groups" and "Lang team Working Groups", additionally the community has also adopted the Working Group terminology for working on areas they are interested in, independent of The Rust Programming Language's Working Groups. It's great that we have a community able to self organise in this way, however it has led to some confusion over who is supporting these efforts, and whether they're _"official"_ Working Groups.

Reviewer: As well as 'lang team working groups' there are WGs for other teams, e.g., dev-tools had a few and I think still does. I think other teams did too.

Author: Yes, initially used lang team working groups as the example as the lang team is most prominent example. I'll make it more general.

[reference-level-explanation]: #reference-level-explanation

### Common Aspects of Working Groups and Project Groups

Before going into the differences between these groups it's important to start with what it is shared between them.

- Both groups have a charter that defines the scope and intent of the group.
- Both groups have a GitHub repository hosted under the rust-lang organization (or separate organisation if necessary) containing the charter and instructions for how community members can monitor the group's progress and/or participate.

> - Groups have at least one shepherd who acts as the leader of the group and is responsible for writing the initial charter, handling adminstrative and communication tasks, as well as delegating responsibilities to other members in the group.

Reviewer 2: This makes it sound a bit like the shepherd is solely responsible for the charter, which probably isn't necessary. In the case of the project-FFI group, it was more collaborative; I think Niko wrote the first draft during a Zoom call. I'm not sure if the charter needs to be mentioned in this bullet at all, actually.

>   - [This blog post][shep-post] explains in detail the role of the shepherds.

Reviewer 3: I think it's "ok" to cite my blog post for now, and I've done it a few times too, but I'd prefer to introduce an RFC or some official document. I don't generally consider my blog posts to be normative or representative of anything but my random thoughts and opinions.

Reviewer 3: I'm also not 100% sure if shepherds are a universal concept -- I think they apply more to project groups. For example, I think the embedded domain WG has its own notion of teams which have shepherds (@jamesmunns can clarify).

Reviewer 3: (Maybe this a terminology point we should address, though -- i.e., the shepherd role is basically a "lead", and maybe we just want to be reserving lead for teams in general?)

- Groups should have a liason member associated with an official Rust team.
  - This liason may also be but is not required to be a shepherd.
- Membership for both groups are represented on the official rust-lang.org website.
- Membership requirements for both groups is decided by the shepherd and should be stated in the charter.
  - Initial membership should try to represent people who have been participating regularly and productively in the respective area.

> - Neither group has _"formal decision making power"_. Where "formal decision making power" is defined as being able to accept RFCs on `rust-lang/rfcs` as well as not having representation in the Core team.

Reviewer 3: This phrasing is confusing to me. Maybe say 
```suggestion
- Neither group has _"formal decision making power"_. Where "formal decision making power" is defined as being able to accept RFCs on `rust-lang/rfcs`. Similarly, neither group has representation on the Core team.
```

  - Groups are of course able to create RFCs as well as advocate their concerns and desired changes to the Rust teams and community.

> - Both groups are entitled to have an space in any or all of Rust's officially managed discussion forums.

Reviewer 2: ```suggestion
- Both groups are entitled to spaces in any or all of Rust's officially managed discussion forums.
```

>   - As the time of this writing this includes [Zulip] and [Discord].

Reviewer 2: I'm not sure if categorizing GitHub as a "discussion forum" is appropriate, but it seems reasonable to assert that groups are entitled to their own repository under the rust-lang organization.

Author: That's already mentioned in the second point?

Reviewer 2: Oh! Yes, sorry.

>   - Groups also free to create their spaces on platforms where the Rust project does not have a official presence. Moderation for these spaces should still follow Rust's [Code of Conduct].

Reviewer 2: ```suggestion
  - Groups are also free to create spaces on platforms where the Rust project does not have a official presence. Moderation for these spaces should still follow Rust's [Code of Conduct].
```

[zulip]: https://rust-lang.zulipchat.com
[discord]: https://discord.gg/rust-lang
[code of conduct]: https://www.rust-lang.org/policies/code-of-conduct
[shep-post]: http://smallcultfollowing.com/babysteps/blog/2019/09/11/aic-shepherds-3-0/

### Working Groups

> Working Group's typically focus on a specific domain of expertise, that is not covered by existing Rust team. Examples of this include Embedded, WebAssembly, and Game Development.

Reviewer 2: ```suggestion
Working Groups typically focus on a specific domain of expertise that is not covered by an existing Rust team. Examples of this include Embedded, WebAssembly, and Game Development.
```

> Working groups are a way to finding new areas that can sustain continual development. Though this is not required if a working group has shown to be regularly productive over a significant period time a working group can become a Rust team.

Reviewer 2: ```suggestion
Working groups are a way of finding new domains that would benefit from continual development. Though it is not required, if a working group has demonstrated consistent productivity over a significant period time, it can become a Rust team.
```
